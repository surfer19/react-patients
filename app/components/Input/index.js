/**
 *
 * Input
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

function Input(props) {
  return (
    <div className="form-group mt-4">
      <label htmlFor="input">{props.label}</label>
      <input
        className="form-control"
        id="input"
        aria-describedby="help"
        placeholder={props.placeholder}
        onChange={e => props.onChange(e.target.value)}
      />
    </div>
  );
}

Input.propTypes = {
  label: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
};

export default Input;
