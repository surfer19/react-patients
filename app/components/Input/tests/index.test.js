/**
 *
 * Tests for Input
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import Input from '../index';
// import 'jest-dom/extend-expect'; // add some helpful assertions
const someLabel = 'someLabel';
const somePlaceholder = 'somePlaceholder';
const renderComponent = (props = {}) =>
  render(<Input label={someLabel} placeholder={somePlaceholder} {...props} />);

describe('<Input />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Input />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('should exist', () => {
    const { container } = renderComponent();
    expect(container.querySelector('input')).not.toBeNull();
  });

  it('should have a class attribute', () => {
    const { container } = renderComponent();
    expect(container.querySelector('input').hasAttribute('class')).toBe(true);
  });

  it('should have a aria-describedby attribute', () => {
    const { container } = renderComponent();
    expect(
      container.querySelector('input').hasAttribute('aria-describedby'),
    ).toBe(true);
  });

  it('should have a placeholder attribute', () => {
    const { container } = renderComponent();
    expect(container.querySelector('input').hasAttribute('placeholder')).toBe(
      true,
    );
  });

  it('should have a label tag', () => {
    const { container } = renderComponent();
    expect(container.querySelector('label')).not.toBe(null);
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Input />);
    expect(firstChild).toMatchSnapshot();
  });
});
