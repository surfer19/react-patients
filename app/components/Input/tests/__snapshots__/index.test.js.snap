// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Should render and match the snapshot 1`] = `
<div
  class="form-group mt-4"
>
  <label
    for="input"
  />
  <input
    aria-describedby="help"
    class="form-control"
    id="input"
  />
</div>
`;
